contrib/cirrus/setup_environment.sh

```diff
@@ -351,7 +350,7 @@ case "$TEST_FLAVOR" in
         slug="gitlab.com/gitlab-org/gitlab-runner"
         helper_fqin="registry.gitlab.com/gitlab-org/gitlab-runner/gitlab-runner-helper:x86_64-latest-pwsh"
         ssh="ssh $ROOTLESS_USER@localhost -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o CheckHostIP=no env GOPATH=$GOPATH"
-        showrun $ssh go get -u github.com/jstemmer/go-junit-report
+        showrun $ssh go install github.com/jstemmer/go-junit-report/v2@v2.0.0
```

Comment:
A comment explaining why this is necessary would be helpful.  Our setup for running the gitlab unit tests was carefully crafted based on explicit instructions from the gitlab runner maintainers.  I don't recall this version-reference being in them.  But in general, our task should simply follow the "latest and greatest" upstream.  If we deviate and it breaks, it can be time-consuming to engage with them (they have other priorities).

Reply:
go get simply does not work with go 1.18, you have to go install.

You could add `@latest` to always use the latest but IMO this will just start breaking on a random PR, we really have to stop doing this and only update versions via PRs. I am tired of having things break on my PRs because of external changes.

Reply:
Furthermore: I cannot find anywhere in that file where it says "this is carefully crafted from anything". If this is fragile code, it must include comments saying so and linking to exact instructions (e.g. on gitlab.com).

Reply:
This is not 'go installing' for podman, it's doing so for the upstream gitlab tests.

> I cannot find anywhere in that file where it says

Hrmmm, I thought I had left a comment about that...Yeah, it's this one:
`# Ref: https://gitlab.com/gitlab-org/gitlab-runner/-/issues/27270#note_499585550`

but you're right, it's not clear at all how specialized this setup is.  I s'pose was unintentionally relying on the sheer volume of **setup** section steps in that link. I'll post a docs PR to make it more clear.

Reply:
IMHO, I think Paul's idea is probably okay, so:

`showrun $ssh go install github.com/jstemmer/go-junit-report/v2@latest`

This tool is only needed at the very end of testing, and AFAIK only used to check and convert the results into a useful (for upstream developers) format.

Reply:
@Luap99's idea is to use a strict version, not `@latest`, and I ferociously agree. I despise using `latest` because, as Paul, states, it always leads to surprise failures at the least convenient time.

Reply:
Normally yes, but this is a different context.  And there's always the `allow_failures: ...` un-comment option since these aren't "our" tests anyway.  That said, we can roll with the fixed version, but if it breaks, somebody will need to work with the gitlab-runner upstream to fix it (same as for any other gitlab test failure I s'pose).